hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -2182,14 +2246,32 @@ int SaraNcpClient::getMtu() {
 int SaraNcpClient::urcs(bool enable) {
     const NcpClientLock lock(this);
     if (enable) {
+        sleepNoPPPWrite_ = false;
         CHECK_TRUE(muxer_.resumeChannel(UBLOX_NCP_AT_CHANNEL) == 0, SYSTEM_ERROR_INTERNAL);
+        if ((ncpId() != PLATFORM_NCP_SARA_R510) ||
+                ((ncpId() == PLATFORM_NCP_SARA_R510) &&
+                !(connState_ == NcpConnectionState::CONNECTED || connState_ == NcpConnectionState::DISCONNECTED))) {
+            muxer_.setKeepAlivePeriod(UBLOX_NCP_KEEPALIVE_PERIOD_DEFAULT);
+        }
         if (ncpId() == PLATFORM_NCP_SARA_U201) {
             // Make sure the modem is responsive again. U201 modems do take a while to
             // go back into functioning state
             CHECK(waitAtResponse(5000, gsm0710::proto::DEFAULT_T2));
         }
     } else {
+        sleepNoPPPWrite_ = true;
+
+        // R510 may be in low power mode, wake it up so we can get our muxer data channel suspend
+        // echo, before we get into sleep and potentially prematurely wake by "network activity" (RX data)
+        if (ncpId() == PLATFORM_NCP_SARA_R510) {
+            CHECK(waitAtResponse(5000, 2000));
+        }
         CHECK_TRUE(muxer_.suspendChannel(UBLOX_NCP_AT_CHANNEL) == 0, SYSTEM_ERROR_INTERNAL);
+
+        // muxer_.suspendChannel(UBLOX_NCP_AT_CHANNEL) does not block muxer AT channel keepalives. All modems will
+        // require this to be disabled to wake on network activity, and not prematurely wake on the echo of the keepalive.
+        muxer_.setKeepAlivePeriod(UBLOX_NCP_KEEPALIVE_PERIOD_R510);
```

Comment:
Nit: this is a bit confusing, as doesn't apply to just R510, might be worth just using `0` here`.

Reply:
Sure, I'll add a `UBLOX_NCP_KEEPALIVE_PERIOD_DISABLED` constant.